exo.core.component.organization.api/src/main/java/org/exoplatform/services/organization/externalstore/IDMExternalStoreImportService.java

```diff
@@ -776,36 +776,13 @@ private UserProfile importUserProfile(String username, boolean deleted) throws E
       LOG.trace("Remove from internal store deleted user profile '{}' from external store", username);
       organizationService.getUserProfileHandler().removeUserProfile(username, true);
     }
-
     UserProfile externalUserProfile = externalStoreService.getEntity(IDMEntityType.USER_PROFILE, username);
-
-    if (externalUserProfile != null && externalUserProfile.getUserInfoMap() != null
-        && !externalUserProfile.getUserInfoMap().isEmpty()) {
-      Map<String, String> externalUserInfoMap = externalUserProfile.getUserInfoMap();
-      Map<String, String> internalUserInfoMap = (internalUserProfile == null
-          || internalUserProfile.getUserInfoMap() == null) ? new HashMap<>() : internalUserProfile.getUserInfoMap();
-      boolean isModified = internalUserInfoMap.isEmpty();
-      if (!isModified) {
-        Set<String> externalUserProfileAttributes = externalUserInfoMap.keySet();
-        for (String externalUserProfileAttribute : externalUserProfileAttributes) {
-          isModified |= !internalUserInfoMap.containsKey(externalUserProfileAttribute)
-              || !Objects.equals(externalUserInfoMap.get(externalUserProfileAttribute),
-                                 internalUserInfoMap.get(externalUserProfileAttribute));
-          if (isModified) {
-            break;
-          }
-        }
-      }
-
-      if (isModified) {
-        internalUserInfoMap.putAll(externalUserInfoMap);
-        UserProfile userProfile = organizationService.getUserProfileHandler().createUserProfileInstance(username);
-        userProfile.setUserInfoMap(internalUserInfoMap);
-        organizationService.getUserProfileHandler().saveUserProfile(userProfile, true);
-        return userProfile;
-      }
+    Map<String, String> userInfoMap = externalUserProfile.getUserInfoMap();
+    if (!userInfoMap.containsKey("username")) {
+      userInfoMap.put("username", username);
     }
-    return internalUserProfile;
+    listenerService.broadcast(IDMExternalStoreService.USER_PROFILE_ADDED_FROM_EXTERNAL_STORE, this, userInfoMap);
+    return null;
```

Comment:
If this function does not have top return a value, we change the signature, otherwise we return the user profile as it was done